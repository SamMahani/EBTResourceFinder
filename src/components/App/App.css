:root {
  --mainDark: #3d3d3f;
  --mainLight: #f6f5f3;
  --mainColorful: #7dce94;
  --whitish: #f9f8fd;
  --borderColor: #ccd1cb;
  --highlight: #a206c9;
}

body {
  background: var(--mainLight);
}

.appWrapper {
  display: grid;
}

.title {
  grid-column: 1 / 3;
  grid-row: 1 / 2;
  font-size: 2em;
  font-weight: 800;
  background: var(--mainColorful);
  color: var(--mainDark);
  padding: 1em 1em 1em 0.5em;
}

.searchWrapper {
  grid-column: 1 / 2;
  grid-row: 2 / 3;
  padding: 1em;
  color: var(--mainDark);
  background-color: var(--whitish);
}

.dropdownWrapper {
  grid-column: 2 / 3;
  grid-row: 2 / 3;
  padding: 1em;
  background-color: var(--whitish);
}

.selectedMarkerWrap {
  grid-column: 1 / 3;
  grid-row: 3 / 4;
  background: var(--mainLight);
  padding: 1em;
  color: var(--mainDark);
}

.googleMapsWrapper {
  grid-column: 1 / 3;
  grid-row: 4/ 10;
}

.googleMapsWrapper div:first-child {
  position: relative;
  width: 100%;
  height: 95vh;
}

.storeListWrapper {
  grid-column: 1 / 3;
  grid-row: 10 / 15;
  padding: 2em 1em;
}
