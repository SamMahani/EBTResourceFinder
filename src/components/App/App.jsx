import "./App.css";
import React from "react";
import axios from "axios";
import StoreList from "../StoreList";
import Search from "../Search";
import DropdownFilter from "../DropdownFilter";
import GoogleMaps from "../GoogleMaps/GoogleMaps.jsx";
import SelectedResource from "../SelectedResource";

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      nearbyStores: [],
      filters: [],
      stores: [],
      markets: [],
      foodbanks: [],
      snapoffices: [],
      wicoffices: [],
      value: "markets",
      currLat: 40.70851,
      currLong: -73.90896
    };
    this.getLocations = this.getLocations.bind(this);
    this.filterStoreByType = this.filterStoreByType.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleLatLongState = this.handleLatLongState.bind(this);
    this.handleSelectedMarker = this.handleSelectedMarker.bind(this);
  }

  componentDidMount() {
    this.getLocations();
  }

  getLocations() {
    const endpoint = `https://www.easyfoodstamps.com/stores?latitude=${
      this.state.currLat
    }&longitude=${this.state.currLong}`;
    axios
      .get(endpoint)
      .then(locations => {
        let nearbyStores = locations.data.stores;
        this.filterStoreByType(nearbyStores);
        this.setState({
          nearbyStores,
          filters: locations.data.filters,
          currLat: this.state.currLat,
          currLong: this.state.currLong
        });
      })
      .catch(err => {
        console.log(err);
      });
  }

  filterStoreByType(nearByStores) {
    let stores = [];
    let markets = [];
    let foodbanks = [];
    let snapoffices = [];
    let wicoffices = [];
    nearByStores.forEach(store => {
      if (store.type === "market") {
        markets.push(store);
      } else if (store.type === "foodbank") {
        foodbanks.push(store);
      } else if (store.type === "snapoffice") {
        snapoffices.push(store);
      } else if (store.type === "wicoffice") {
        wicoffices.push(store);
      } else {
        stores.push(store);
      }
    });
    this.setState({
      stores,
      markets,
      foodbanks,
      snapoffices,
      wicoffices
    });
  }

  handleChange(e) {
    this.setState({ value: e.target.value });
  }

  handleLatLongState(lat, lng) {
    this.setState({
      currLat: lat,
      currLong: lng
    });
  }

  handleSelectedMarker(selectedMarker) {
    this.setState({
      selectedMarker
    });
  }

  render() {
    return (
      <div className="appWrapper">
        <header data-testid="titleTest" className="title">
          EBT Resource Finder
        </header>
        <div className="searchWrapper">
          <Search
            getLocations={this.getLocations}
            handleLatLongState={this.handleLatLongState}
          />
        </div>
        <div className="dropdownWrapper">
          <DropdownFilter
            value={this.state.value}
            handleChange={this.handleChange}
          />
        </div>
        <section className="selectedMarkerWrap">
          <SelectedResource selectedMarker={this.state.selectedMarker} />
        </section>
        <section className="googleMapsWrapper">
          <GoogleMaps
            handleSelectedMarker={this.handleSelectedMarker}
            list={this.state[this.state.value]}
            currLat={this.state.currLat}
            currLong={this.state.currLong}
          />
        </section>
        <div className="storeListWrapper">
          <StoreList list={this.state[this.state.value]} />
        </div>
      </div>
    );
  }
}

export default App;
