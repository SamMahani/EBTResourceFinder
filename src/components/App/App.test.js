import React from "react";
import { render, cleanup } from "react-testing-library";
import "jest-dom/extend-expect";
import App from "./index.js";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
