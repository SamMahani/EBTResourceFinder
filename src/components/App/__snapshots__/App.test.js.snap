// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="appWrapper"
  >
    <header
      class="title"
      data-testid="titleTest"
    >
      EBT Resource Finder
    </header>
    <div
      class="searchWrapper"
    >
      <form>
        <label
          class="labelWrapper"
        >
          Zip: 
          <input
            class="inputBar"
            name="zip"
            type="text"
            value=""
          />
        </label>
        <input
          class="submitButton"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
    <div
      class="dropdownWrapper"
    >
      <select
        class="dropdown"
      >
        <option
          value="markets"
        >
          market
        </option>
        <option
          value="foodbanks"
        >
          foodbank
        </option>
        <option
          value="snapoffices"
        >
          snapoffice
        </option>
        <option
          value="wicoffices"
        >
          wicoffice
        </option>
        <option
          value="stores"
        >
          store
        </option>
      </select>
    </div>
    <section
      class="selectedMarkerWrap"
    >
      <section />
    </section>
    <section
      class="googleMapsWrapper"
    >
      <div>
        Loading...
      </div>
    </section>
    <div
      class="storeListWrapper"
    >
      <section />
    </div>
  </div>
</DocumentFragment>
`;
